cpp/refactor_module/algorithm/refactor.cpp

```diff
@@ -863,3 +863,129 @@ void Refactor::RenameType(mgp_list *args, mgp_graph *memgraph_graph, mgp_result
     return;
   }
 }
+
+void Refactor::MergeNodes(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::MemoryDispatcherGuard guard{memory};
+  const auto arguments = mgp::List(args);
+  auto graph = mgp::Graph(memgraph_graph);
+  const auto record_factory = mgp::RecordFactory(result);
+
+  try {
+    const auto nodes = arguments[0].ValueList();
+    if (nodes.Empty()) {
+      throw mgp::ValueException(std::string(kMergeNodesEmptyListError).c_str());
```

Comment:
I think you can use kMergeNodesEmptyListError.data() instead of converting to string and then c_str

Reply:
or try even without data(), I can think we can avoid temporary string allocation